### PR TITLE
fix: fix method name and add unit tests

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
If the added resource_reference in a field is a child_type, we will search in the database to see if the resource is existing. If not, it results in breaking change, otherwise non-breaking.

### CODE REVIEW

File: test/comparator/test_field_comparator.py [hunk 1]
```diff
@@ -321,6 +321,37 @@ def test_resource_reference_addition_non_breaking(self):
         self.assertEqual(finding.category.name, "RESOURCE_REFERENCE_ADDITION")
         self.assertEqual(finding.change_type.name, "MINOR")
 
+    def test_resource_reference_addition_non_breaking2(self):
```
Comment (arithmetic1728) [inline test/comparator/test_field_comparator.py:324]: Maybe change the test name to indicate this tests child type resource ref additions.
```suggestion
    def test_resource_reference_child_type_addition_non_breaking(self):
```

File: test/comparator/test_field_comparator.py [hunk 2]
```diff
@@ -294,7 +294,7 @@ def test_resource_reference_addition_breaking(self):
             "A resource reference option is added to the field `Test`, but it is not defined anywhere",
         )
 
-    def test_resource_reference_addition_non_breaking(self):
+    def test_resource_reference_addition_non_breaking1(self):
```
Comment (arithmetic1728) [inline test/comparator/test_field_comparator.py:297]: If we change the test name of the other test, this name can stay the same
```suggestion
    def test_resource_reference_addition_non_breaking(self):
```

**Review by arithmetic1728 — APPROVED** [review]

File: test/comparator/test_field_comparator.py (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply test/comparator/test_field_comparator.py:324]: Thanks! I have concerns to make the test name too long, but it makes sense to indicate the intentions for a specific unit test.

Comment (xiaozhenliu-gg5) [reply test/comparator/test_field_comparator.py]: Done.

File: test/comparator/test_field_comparator.py (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply test/comparator/test_field_comparator.py]: Done.

**Pull request merged**